src/core/api/link_api.cpp

```diff
@@ -418,6 +429,17 @@ otError otLinkCslSetPeriod(otInstance *aInstance, uint16_t aPeriod)
     return error;
 }
 
+otError otLinkCslSetPeriodMs(otInstance *aInstance, uint32_t aPeriod)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit((aPeriod == 0 || aPeriod * 1000 / kUsPerTenSymbols <= 0xFFFF), error = kErrorInvalidArgs);
```

Comment:
Why check `aPreiod ==0` separately? (can remove and include with the other check?)


Use `NumericLimits<uint16_t>::kMax` instead of `0xFFFF`?